website/docs/pipelines/promoting-applications.mdx

```diff
@@ -185,13 +202,97 @@ type: Opaque
+      - "app-promotion-credentials"
+    resources:
+      - "secrets"
+    verbs:
+      - "get"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: pipeline-controller-read-app-promotion-credentials
+  namespace: app
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: read-app-promotion-credentials
+subjects:
+  - kind: ServiceAccount
+    name: pipeline-controller
+    namespace: pipeline-system
+```
+</details>
+
+3. **Do not use long-live tokens**: set an expiration date and rotate them according to your security policy.
+
+<details><summary>Expand to see example</summary>
+
+![create token witih expiration](img/create-token-with-expiration.png)
+</details>
+
+4. **Honour the least privilege principle**: avoid having high privilege tokens. Restrict the token to your just your repo and to just the operations required.
+
+<details><summary>Expand to see example</summary>
+
+For example, if the case of GitHub, use [fine-grained tokens](https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/) to only
+allow access to the single repo that your configuration manifests exist.
+
+![create token least priviledge](img/fine-grained-token.png)
+ </details>
+
+5. **Review active access tokens on a regular basis**: to ensure that only the ones that are required are present at all times.
+
+<details><summary>Expand to see example</summary>
+
+For example, using github and fine-grained tokens you [could do so](https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/#approving-and-auditing-personal-access-tokens).
+
+![review tokens](img/manage-fine-grained.png)
+</details>
+
+6. **Reduce number of secrets to maintain**: Whenever ends in a more secure setup, favor git over https with tokens instead of ssh and deploy keys. Given that a token
+is required for git provider api operations, to use git over ssh would require to secure and manage the deploy keys, as well as the token.
+In addition, tokens have an expiration date (so requires rotation) and could have finer access control (to just allow the pipeline repo).
+
+7. **Review git provider recommendations and examples**
+- [GitHub](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token)
+- [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html)
+
 ### Define promotion in pipeline resource
 
 The field `.spec.promotion.strategy.pull-request` defines details about the Git repository used for promoting the given app. Set the `secretRef.name` field to the name of the Secret created in the previous step and the `url` and `branch` fields to the Git repository's URL and optionally a specific branch (if the branch is not set, it defaults to `main`). If using the `generic-hmac` Provider from leaf clusters, also set the `.spec.promotion.strategy.secretRef.name` to the name of the Secret created previously.
```

Comment:
Why does this default to `main` when Flux source controller defaults to `master` https://github.com/fluxcd/source-controller/blob/main/docs/spec/v1beta1/gitrepositories.md?plain=1#L35-L36 ?

To be clear, I think it should be `main` per https://github.com/github/renaming#new-repositories-use-main-as-the-default-branch-name but it feels confusing that we have two different defaults?

Reply:
Actually as of https://github.com/weaveworks/pipeline-controller/pull/160 the base branch is now configurable so we need to update the docs.

Reply:
We can also make that a required field and avoid setting a default value. We can't change Flux's default branch name and no point setting it to a different one by default. I'll add an issue for that and to update the docs.

Reply:
👍